pkg/storage/utils.go

```diff
@@ -0,0 +1,117 @@
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get PVC names for selector %v", selector)
+	}
+	numPVCs := len(pvcs)
+	if numPVCs != 1 {
+		return "", fmt.Errorf("expected exactly one PVC attached to storage %v, but got %v, %v", storageName, numPVCs, pvcs)
+	}
+	return pvcs[0], nil
+}
+
+// GetMachineReadableFormatForList gives machine-readable StorageList definition
+func GetMachineReadableFormatForList(storage []Storage) StorageList {
+	return StorageList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "List",
+			APIVersion: apiVersion,
+		},
+		ListMeta: metav1.ListMeta{},
+		Items:    storage,
+	}
+}
+
+// GetMachineReadableFormat gives machine-readable Storage definition
+// storagePath indicates the path to which the storage is mounted to, "" if not mounted
+func GetMachineReadableFormat(storageName, storageSize, storagePath string) Storage {
+	return Storage{
+		TypeMeta:   metav1.TypeMeta{Kind: "storage", APIVersion: apiVersion},
+		ObjectMeta: metav1.ObjectMeta{Name: storageName},
+		Spec: StorageSpec{
+			Size: storageSize,
+			Path: storagePath,
+		},
+	}
+}
+
+// GetMachineFormatWithContainer gives machine-readable Storage definition
+// storagePath indicates the path to which the storage is mounted to, "" if not mounted
+func GetMachineFormatWithContainer(storageName, storageSize, storagePath string, container string) Storage {
+	storage := Storage{
+		TypeMeta:   metav1.TypeMeta{Kind: "storage", APIVersion: apiVersion},
+		ObjectMeta: metav1.ObjectMeta{Name: storageName},
+		Spec: StorageSpec{
+			Size: storageSize,
+			Path: storagePath,
+		},
+	}
+	storage.Spec.ContainerName = container
+	return storage
+}
+
+// generatePVCName generates a PVC name from the Devfile volume name, component name and app name
+func generatePVCName(volName, componentName, appName string) (string, error) {
+
+	pvcName, err := util.NamespaceKubernetesObject(volName, componentName)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	pvcName, err = util.NamespaceKubernetesObject(pvcName, appName)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	return pvcName, nil
+}
+
+// ConvertListLocalToMachine It converts storage config list to StorageList type
+func ConvertListLocalToMachine(storageListConfig []localConfigProvider.LocalStorage) StorageList {
+
+	var storageListLocal []Storage
+
+	for _, storeLocal := range storageListConfig {
+		s := GetMachineReadableFormat(storeLocal.Name, storeLocal.Size, storeLocal.Path)
+		s.Spec.ContainerName = storeLocal.Container
+		storageListLocal = append(storageListLocal, s)
+	}
+
+	return GetMachineReadableFormatForList(storageListLocal)
+}
+
+// MachineReadableSuccessOutput outputs a success output that includes
+// storage information
+func MachineReadableSuccessOutput(storageName string, message string) {
+	machineOutput := machineoutput.GenericSuccess{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "storage",
```

Comment:
You might need to make it inline with #4948 where 'storage' → 'Storage'.

Reply:
Fixed

Reply:
>Fixed

Did you forget to push?

Reply:
Fixed